### PR TITLE
framework/st_things : collection feature was separated.

### PR DESCRIPTION
**Pull request by Hyunjun85**
It needs when product commercialization.
So developer should select the feature activation.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [eb1e3bf7fe60d1dd40fa976afab6bd9c3eed9714] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [eb1e3bf7fe60d1dd40fa976afab6bd9c3eed9714] - Code Rule Check OK.

File: framework/src/st_things/st_things_request_handler_collection.c [hunk 1]
```diff
@@ -1067,3 +1069,4 @@ int handle_post_req_on_collection_rsrc(things_resource_s *collection_rsrc)
 
 	return result ? 1 : 0;
 }
+#endif
```
Comment (JeongJunSik) [inline framework/src/st_things/st_things_request_handler_collection.c]: new line

File: framework/src/st_things/st_things_request_handler_collection.c [hunk 2]
```diff
@@ -27,6 +27,8 @@
 
 #define TAG "[st_things_sdk]"
 
+#ifdef CONFIG_ST_THINGS_COLLECTION
```
Comment (JeongJunSik) [inline framework/src/st_things/st_things_request_handler_collection.c]: this file is not build.
it's not need this def

**Pull request closed without merging**